index.js

```diff
@@ -18,34 +18,31 @@ var typedArrays = {
 };
 
 var slice = String.prototype.slice;
-var toStrTags = {};
 var gOPD = Object.getOwnPropertyDescriptor;
-if (hasToStringTag && gOPD && Object.getPrototypeOf) {
-	forEach(typedArrays, function (_, typedArray) {
-		var arr = new global[typedArray]();
-		if (!(Symbol.toStringTag in arr)) {
-			throw new EvalError('this engine has support for Symbol.toStringTag, but ' + typedArray + ' does not have the property! Please report this.');
-		}
-		var proto = Object.getPrototypeOf(arr);
-		var descriptor = gOPD(proto, Symbol.toStringTag);
-		if (!descriptor) {
-			var superProto = Object.getPrototypeOf(proto);
-			descriptor = gOPD(superProto, Symbol.toStringTag);
-		}
-		toStrTags[typedArray] = descriptor.get;
-	});
-}
-
 var tryTypedArrays = function tryTypedArrays(value) {
-	var anyTrue = false;
-	forEach(toStrTags, function (getter, typedArray) {
-		if (!anyTrue) {
-			try {
-				anyTrue = getter.call(value) === typedArray;
-			} catch (e) { /**/ }
+	if (Object.getPrototypeOf) {
+		var typedArray, arr, proto, descriptor;
+		/* jscs:disable disallowNodeTypes */
+		for (typedArray in typedArrays) {
+			if (has(typedArrays, typedArray)) {
+				arr = new global[typedArray]();
+				if (!(Symbol.toStringTag in arr)) {
+					throw new EvalError('this engine has support for Symbol.toStringTag, but ' + typedArray + ' does not have the property! Please report this.');
+				}
+				proto = Object.getPrototypeOf(arr);
+				descriptor = gOPD(proto, Symbol.toStringTag);
+				if (!descriptor) {
+					proto = Object.getPrototypeOf(proto);
+					descriptor = gOPD(proto, Symbol.toStringTag);
+				}
+				if (descriptor) {
+					return typedArrays[descriptor.get.call(Object(value))];
+				}
+			}
 		}
-	});
-	return anyTrue;
+	}
+	/* jscs:enable disallowNodeTypes */
+	return false;
```

Comment:
wouldn't this mean that in a browser without `Object.getPrototypeOf`, this will always return false? This module needs to work in pre-ES5 browsers that have TypedArrays.